filterapi/filterconfig.go

```diff
@@ -224,6 +226,16 @@ type AzureAuth struct {
 	AccessToken string `json:"accessToken"`
 }
 
+// GCPAuth defines the file containing GCP credential that will be mounted to the external proc.
+type GCPAuth struct {
+	// AccessToken is the access token as a literal string.
+	AccessToken string `json:"accessToken"`
+	// Region is the GCP region to use for the request.
+	Region string `json:"region"`
+	// ProjectName is the GCP project name to use for the request.
+	ProjectName string `json:"projectName"`
```

Comment:
```suggestion
	// ProjectName is the GCP project id to use for the request.
	ProjectID string `json:"projectID"`
```

Reply:
Apparently this needs to be project ID and not project name.
[docs](https://cloud.google.com/iam/docs/service-accounts-create#creating)

The service account's email address in gcp follow below format
SERVICE_ACCOUNT_NAME@PROJECT_ID.iam.gserviceaccount.com

we use the service-account email when getting the token